client/state/plugins/installed/selectors-ts.ts

```diff
@@ -95,3 +95,55 @@ export const getAllPluginsIndexedBySiteId = createSelector(
 	},
 	( state ) => [ getAllPluginsIndexedByPluginSlug( state ), getSiteIdsThatHavePlugins( state ) ]
 ) as { ( state: AppState ): { [ siteId: number ]: { [ pluginSlug: string ]: Plugin } } };
+
+export function getPluginsOnSites( state: AppState, plugins: Plugin[] ) {
+	return Object.values( plugins ).reduce(
+		( acc: { [ pluginSlug: string ]: Plugin }, plugin: Plugin ) => {
+			const siteIds = Object.keys( plugin.sites ).map( Number );
+			const pluginOnSites = getPluginOnSites( state, siteIds, plugin.slug );
+			if ( pluginOnSites ) {
+				acc[ plugin.slug ] = pluginOnSites;
```

Comment:
Do we really want to have arrays with `undefined` in them? How is this being handled at the component level afterwards?

Reply:
This is how it worked before, and I tried to preserve the existing behavior of each selector apart from putting the site dependent properties on site objects instead of the plugin object. The components would need to be investigated to see how they would handle it.

Reply:
I'd very much like to see why the components consider an array of `undefined` as expected in this case. Could you provide more details on where and how it worked that way, and why? Thanks in advance!